pkg/testexporter/correctness/simple.go

```diff
@@ -86,6 +125,80 @@ func (tc *simpleTestCase) Query(ctx context.Context, client v1.API, selectors st
 	return result, nil
 }
 
-func (tc *simpleTestCase) Quantized(duration time.Duration) time.Duration {
-	return duration.Truncate(time.Minute)
+func (tc *simpleTestCase) Test(ctx context.Context, client v1.API, selectors string, start time.Time, duration time.Duration) (bool, error) {
+	log := spanlogger.FromContext(ctx)
+	pairs, err := tc.Query(ctx, client, selectors, start, duration)
+	if err != nil {
+		level.Info(log).Log("err", err)
+		return false, err
+	}
+
+	return verifySamples(ctx, tc, pairs, duration, tc.cfg), nil
+}
+
+func (tc *simpleTestCase) MinQueryTime() time.Time {
+	return calculateMinQueryTime(tc.cfg.durationQuerySince, tc.cfg.timeQueryStart)
+}
+
+func verifySamples(ctx context.Context, tc Case, pairs []model.SamplePair, duration time.Duration, cfg CommonTestConfig) bool {
```

Comment:
The code would mostly be the same for both. Technically the condition [here](https://github.com/cortexproject/cortex/pull/2595/files#diff-a0503a270660e1d594402bfeec0030ceR159) could also be false for `simpleTestCase` as well because the minimum query window is configurable. I think we should not bother having a hard requirement for very short windows because a network glitch or rollout could easily cause us to miss 1 or 2 samples.